addons/mail/static/src/models/attachment_viewer_item.js

```diff
@@ -0,0 +1,170 @@
+/** @odoo-module **/
+
+import { registerModel } from "@mail/model/model_core";
+import { attr, one } from "@mail/model/model_field";
+
+registerModel({
+    name: "AttachmentViewerItem",
+    identifyingFields: [["attachmentOwner"]],
```

Comment:
Maybe rename `AttachmentViewerViewable`.

I feel like the "item" concept should be strongly related to the parent model, whereas here it's just related to the attachment/document generally speaking.

Each attachment/document is a `AttachmentViewerViewable` regardless of it's actually being viewed at a point in time.